pkg/math/oracle/utils.go

```diff
@@ -55,3 +56,19 @@ func (m *MedianAggregator) GetProviderPrice(
 
 	return scaledPrice, nil
 }
+
+// UpdateMarketMap updates the market map for the oracle.
+func (m *MedianAggregator) UpdateMarketMap(marketMap mmtypes.MarketMap) {
+	m.Lock()
+	defer m.Unlock()
+
+	m.cfg = marketMap
+}
+
+// GetMarketMap returns the market map for the oracle.
+func (m *MedianAggregator) GetMarketMap() mmtypes.MarketMap {
+	m.Lock()
+	defer m.Unlock()
+
+	return m.cfg
```

Comment:
This might mean that this object is copied a bunch which we probably dont want to be doing

Reply:
gonna use a pointer ref instead